lib/msf/core/rpc/v10/rpc_session.rb

```diff
@@ -467,13 +467,17 @@ def rpc_meterpreter_directory_separator(sid)
   def rpc_compatible_modules( sid)
     ret = []
 
-    mtype = "post"
-    names = self.framework.post.module_refnames.map{ |x| "post/#{x}" }
-    names.each do |mname|
-      m = _find_module(mtype, mname)
-      next if not m.session_compatible?(sid)
-      ret << m.fullname
+    mtypes = ['post', 'auxiliary']
```

Comment:
Maybe this would be something like

```ruby
# New cached module searching logic for aux modules
session_type = self.framework.sessions[sid].type
search_params = { 'session_type' => [[session_type], []], 'type' => [['auxiliary'], []] }
auxiliary_modules = Msf::Modules::Metadata::Cache.instance.find(search_params)

# Old logic for post modules
post_modules = []
mtype = "post"
names = self.framework.post.module_refnames.map{ |x| "post/#{x}" }
names.each do |mname|
  m = _find_module(mtype, mname)
  next if not m.session_compatible?(sid)
  post_modules << m.fullname
end

{ "modules" => auxiliary_modules + post_modules }
```